src/main/kotlin/lotto/domain/LottoGame.kt

```diff
@@ -0,0 +1,27 @@
+package lotto.domain
+
+import java.math.BigInteger
+
+class LottoGame(private val lottos: List<Lotto>, private val winningLotto: Lotto) {
+
+    private val prize: BigInteger =
+        MATCH3_PRIZE.toBigInteger().multiply(match(3).toBigInteger()) + MATCH4_PRIZE.toBigInteger()
+            .multiply(match(4).toBigInteger()) + MATCH5_PRIZE.toBigInteger()
+            .multiply(match(5).toBigInteger()) + MATCH6_PRIZE.toBigInteger().multiply(match(6).toBigInteger())
+
+    val profit: Double = prize.toDouble() / (lottos.size * LOTTO_PRICE)
+
+    fun match(i: Int): Int {
+        return lottos.filter { lotto ->
+            lotto.match(winningLotto) == i
+        }.size
```

Comment:
코틀린에는 교집합을 구할 수 있는 함수를 제공하고 있습니다! 사용해보시면 좋을것 같네요~!
```suggestion
        return this.numbers.intersect(lotto.numbers).size
```